components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/UserResolveExecutor.java

```diff
@@ -0,0 +1,201 @@
+        if (usernameClaim == null) {
+            executorResponse = new ExecutorResponse(STATUS_USER_INPUT_REQUIRED);
+        } else {
+            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(usernameClaim);
+            executorResponse = resolveUser(tenantAwareUsername, context.getTenantDomain(), context);
+        }
+        return executorResponse;
+    }
+
+    /**
+     * Resolves user attributes and adds them to the flow context.
+     *
+     * @param username     Username to resolve.
+     * @param tenantDomain Tenant domain.
+     * @param context      Flow execution context.
+     */
+    private ExecutorResponse resolveUser(String username, String tenantDomain, FlowExecutionContext context) {
+
+        ExecutorResponse executorResponse;
+        try {
+            // Obtain the realm service and user realm for the tenant.
+            RealmService realmService = IdentifierAuthenticatorServiceComponent.getRealmService();
+            int tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
+            UserRealm userRealm = (UserRealm) realmService.getTenantUserRealm(tenantId);
+
+            if (userRealm == null) {
+                throw new AuthenticationFailedException(
+                        IdentifierErrorConstants.ErrorMessages.CANNOT_FIND_THE_USER_REALM_FOR_THE_GIVEN_TENANT
+                                .getCode(), String.format(
+                        IdentifierErrorConstants.ErrorMessages.CANNOT_FIND_THE_USER_REALM_FOR_THE_GIVEN_TENANT
+                                .getMessage(), tenantId),
+                        User.getUserFromUserName(username));
+            }
+
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            String resolvedUsername = resolveQualifiedUsername(username, userStoreManager);
+            Claim[] claims = userStoreManager.getUserClaimValues(resolvedUsername, null);
+            if (claims != null && claims.length > 0) {
+                Map<String, String> claimMap = Arrays.stream(claims)
+                        .filter(c -> c != null && c.getClaimUri() != null)
+                        .collect(Collectors.toMap(Claim::getClaimUri, Claim::getValue));
+                context.getFlowUser().addClaims(claimMap);
+            }
+            executorResponse = new ExecutorResponse(STATUS_COMPLETE);
+
+        } catch (UserStoreException e) {
+            log.debug("Error fetching attributes for: " +
+                    LoggerUtils.getMaskedContent(username) + " in tenant: " + tenantDomain, e);
+            executorResponse = new ExecutorResponse(STATUS_ERROR);
+        } catch (Exception e) {
+            log.debug("Unexpected error while fetching attributes for: " +
+                    LoggerUtils.getMaskedContent(username) + " in tenant: " + tenantDomain, e);
+            executorResponse = new ExecutorResponse(STATUS_ERROR);
+        }
```

Comment:
Both of these catch blocks seems to be handling the errors in more or less the same way. Can combine in to a single block. Make sure to use specific errors.